services/roomService/src/types/Player.test.ts

```diff
@@ -0,0 +1,26 @@
+import { MessageType } from '../CoveyTypes';
+import { createMessageForTesting, createPlayerForTesting } from '../TestUtils';
+
+describe('Player', () => {
+  it('has an empty town message chain upon initialization', () => {
+    const player = createPlayerForTesting();
+    expect(player.townMessageChain.messages.length).toBe(0);
+  });
+
+  describe('receiveMessage', () => {
+    it('adds an incoming townwide message to the town message list', () => {
+      const player = createPlayerForTesting();
+      const message = createMessageForTesting(MessageType.TownMessage, player);
+      player.receiveMessage(message);
+      const townMessages = player.townMessageChain.messages;
+      expect(townMessages.length).toBe(1);
+      expect(townMessages[0]).toBe(message);
```

Comment:
fun fact: you can just use `expect(townMessages).toContain(message)`, especially if you already checked the length. not something you need to change but it just makes it more readable